src/OpenLoco/Paint/Paint.cpp

```diff
@@ -41,10 +42,128 @@ namespace OpenLoco::Paint
         return &_session;
     }
 
+    void registerHooks()
+    {
+        registerHook(
+            0x004622A2,
+            [](registers& regs) -> uint8_t {
+                registers backup = regs;
+
+                PaintSession session;
+                session.generate();
+
+                regs = backup;
+                return 0;
+            });
+    }
+
+    // 0x00461CF8
+    static void tilePaintSetup(PaintSession& session, const Map::map_pos& loc)
+    {
+        registers regs{};
+        regs.eax = loc.x;
+        regs.ecx = loc.y;
+        call(0x00461CF8, regs);
+    }
+
+    // 0x004617C6
+    static void tilePaintSetup2(PaintSession& session, const Map::map_pos& loc)
+    {
+        registers regs{};
+        regs.eax = loc.x;
+        regs.ecx = loc.y;
+        call(0x004617C6, regs);
+    }
+
+    // 0x0046FA88
+    static void entityPaintSetup(PaintSession& session, const Map::map_pos& loc)
+    {
+        registers regs{};
+        regs.eax = loc.x;
+        regs.ecx = loc.y;
+        call(0x0046FA88, regs);
+    }
+
+    // 0x0046FB67
+    static void entityPaintSetup2(PaintSession& session, const Map::map_pos& loc)
+    {
+        registers regs{};
+        regs.eax = loc.x;
+        regs.ecx = loc.y;
+        call(0x0046FB67, regs);
+    }
+
+    template<uint8_t rotation>
+    void generateRotated(PaintSession& session)
```

Comment:
Wouldn't it make more sense to have this as a class method? Passing in the 'this' feels kinda C-like.

Reply:
I didn't want to pollute the header file with lots of things but maybe there is a way of doing it.

Reply:
Okay I have a solution: https://gist.github.com/duncanspumpkin/bd5be296703e92f050336e16eaf8afcb what do you think? It makes the part that needs to be a template still a free function but the rest of it becomes a member function.

Reply:
I think this does improve the readability of the code 👍